assets/src/edit-story/components/inspector/documentInspector.js

```diff
@@ -1,9 +1,187 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+import styled, { css } from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../app/story';
+import { useConfig } from '../../app/config';
+import UploadButton from '../uploadButton';
+import useInspector from './useInspector';
+import { SelectMenu, InputGroup } from './shared';
+
+const ButtonCSS = css`
+	color: ${ ( { theme } ) => theme.colors.mg.v1 };
+	font-size: 14px;
+	width: 100%;
+	padding: 15px;
+	background: none;
+	margin: 5px 0px;
+`;
+const Img = styled.img`
+	width: 100%;
+	max-height: 300px
+`;
+
+const Group = styled.div`
+	border-color: ${ ( { theme } ) => theme.colors.mg.v1 };
+	display: block;
+	align-items: center;
+	margin: 15px 0px;
+`;
+
+const RemoveButton = styled.button`
+	${ ButtonCSS }
+`;
 
 function DocumentInspector() {
+	const {
+		actions: { loadStatuses, loadUsers, setDisabledStatuses },
+		state: { users, statuses, disabledStatuses },
+	} = useInspector();
+
+	const {
+		state: { meta: { isSaving }, story: { author, status, slug, date, excerpt, featuredMediaUrl, password }, capabilities },
+		actions: { updateStory, deleteStory },
+	} = useStory();
+
+	const { postThumbnails } = useConfig();
+
+	useEffect( () => {
+		loadStatuses();
+		loadUsers();
+	} );
+
+	useEffect( () => {
+		const states = ( status === 'future' ) ? [ 'pending' ] : [ 'future', 'pending' ];
+		setDisabledStatuses( states );
+	}, [ status, setDisabledStatuses ] );
+
+	const handleChangeStatus = useCallback(
+		( evt ) => 	updateStory( { properties: { status: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeAuthor = useCallback(
+		( evt ) => updateStory( { properties: { author: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangePassword = useCallback(
+		( value, evt ) => updateStory( { properties: { password: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeDate = useCallback(
+		( value, evt ) => updateStory( { properties: { date: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeExcerpt = useCallback(
+		( value, evt ) => updateStory( { properties: { excerpt: evt.target.value } } ),
+		[ updateStory ],
+	);
+
+	const handleChangeSlug = useCallback(
+		( value, evt ) => updateStory( { properties: { slug: evt.target.value } } ),
+		[ updateStory ],
+	);
```

Comment:
These can all be generalised to:

```js
	const handleChangeValue = useCallback(
		( prop ) => ( value, evt ) => updateStory( { properties: { [ prop ]: evt.target.value } } ),
		[ updateStory ],
	);
```

And then further down for the individual input:

```js
		onChange={ handleChangeValue( 'slug' ) }
```

Reply:
Good catch. Fixed in https://github.com/ampproject/amp-wp/pull/4122/commits/9c4c650652a0962d34eb04c0b0f682773f412aa8